src/csharp/Microsoft.Spark/Sql/RowConstructor.cs

```diff
@@ -4,112 +4,49 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using Microsoft.Spark.Sql.Types;
 using Razorvine.Pickle;
 
 namespace Microsoft.Spark.Sql
 {
     /// <summary>
-    /// RowConstructor is a custom unpickler for GenericRowWithSchema in Spark.
+    /// RowConstructorConstructor is a custom unpickler for GenericRowWithSchema in Spark.
     /// Refer to spark/sql/core/src/main/scala/org/apache/spark/sql/execution/python/
     /// EvaluatePython.scala how GenericRowWithSchema is being pickled.
     /// </summary>
-    internal sealed class RowConstructor : IObjectConstructor
+    internal sealed class RowConstructorConstructor : IObjectConstructor
     {
         /// <summary>
         /// Cache the schemas of the rows being received. Multiple schemas may be
         /// sent per batch if there are nested rows contained in the row. Note that
-        /// this is thread local variable because one RowConstructor object is
+        /// this is thread local variable because one RowConstructorConstructor object is
         /// registered to the Unpickler and there could be multiple threads unpickling
-        /// the data using the same object registered.
+        /// the data using the same registered object.
         /// </summary>
         [ThreadStatic]
         private static IDictionary<string, StructType> s_schemaCache;
 
         /// <summary>
-        /// The RowConstructor that created this instance.
+        /// Used by Unpickler to pass unpickled schema for handling.
         /// </summary>
-        private readonly RowConstructor _parent;
-
-        /// <summary>
-        /// Stores the args passed from construct().
-        /// </summary>
-        private readonly object[] _args;
-
-        public RowConstructor() : this(null, null)
-        {
-        }
-
-        public RowConstructor(RowConstructor parent, object[] args)
-        {
-            _parent = parent;
-            _args = args;
-        }
-
-        /// <summary>
-        /// Used by Unpickler to pass unpickled data for handling.
-        /// </summary>
-        /// <param name="args">Unpickled data</param>
-        /// <returns>New RowConstructor object capturing args data</returns>
+        /// <param name="args">Unpickled schema</param>
+        /// <returns>New RowConstructor object capturing the schema.</returns>
         public object construct(object[] args)
         {
-            // Every first call to construct() contains the schema data. When
-            // a new RowConstructor object is returned from this function,
-            // construct() is called on the returned object with the actual
-            // row data. The original RowConstructor object may be reused by the
-            // Unpickler and each subsequent construct() call can contain the
-            // schema data or a RowConstructor object that contains row data.
```

Comment:
this comment is not relevant any more? (If not, can you update instead of removing?)

Reply:
updated class summary.